inc/blocks/namespace.php

```diff
@@ -157,6 +160,90 @@ function on_save_post( int $post_ID, WP_Post $post, bool $update ) : void {
 	}
 }
 
+/**
+ * Synchronise any XBs on the widget page with a shadow post type.
+ *
+ * @param array $instance The current widget settings.
+ * @param array $new_instance The of new widget settings.
+ * @param array $old_instance The of old widget settings.
+ * @param object $widget The current widget instance.
+ * @return array $instance The current widget settings.
+ */
+function on_widgets_save( $instance, $new_instance, $old_instance, $widget ) {
+
+	if ( ! isset( $instance['content'] ) ) {
+		return $instance;
+	}
+
+	// Scan for XBs in the post content.
+	$blocks = parse_blocks( $instance['content'] );
+	$xbs = find_xbs( $blocks );
```

Comment:
In the other function there were other early returns if no XB's were found, if we can avoid doing unnecessary DB queries then we should.